geoh5py/ui_json/input_file.py

```diff
@@ -0,0 +1,571 @@
+#  Copyright (c) 2021 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+
+from __future__ import annotations
+
+import json
+import os
+import warnings
+from copy import deepcopy
+from typing import Any, Callable
+from uuid import UUID
+
+import numpy as np
+
+from geoh5py.groups import ContainerGroup
+from geoh5py.workspace import Workspace
+
+from .constants import ui_validations
+from .constants import validations as default_validations
+from .validators import InputValidators
+
+
+class InputFile:
+    """
+    Handles loading ui.json input files.
+
+    Attributes
+    ----------
+    filepath : Path to input file.
+    data : Input file content parsed to flat dictionary of key:value.
+    ui_json: User interface serializable as ui.json format
+
+    Methods
+    -------
+    default()
+        Defaults values in 'data' attribute to those stored in default_ui 'default' fields.
+    write_ui_json()
+        Writes a ui.json formatted file from 'data' attribute contents.
+    read_ui_json()
+        Reads a ui.json formatted file into 'data' attribute dictionary.  Optionally filters
+        ui.json fields other than 'value'.
+
+    """
+
+    def __init__(
+        self,
+        data: dict[str, Any] = None,
+        filepath: str | None = None,
+        ui_json: dict[str, Any] = None,
+        validations: dict = None,
+        workspace: Workspace = None,
+    ):
```

Comment:
Can we get a validate=True, and a validator_opts={} argument here to help with testing incomplete input files?  validate=False should ignore all the validations, and validate=True with validator_opts = {"ignore_requirements": True} should run all the validations besides the requirements checker.

Reply:
Done